nni/tools/nnictl/common_utils.py

```diff
@@ -95,3 +97,23 @@ def generate_folder_name():
         temp_dir = generate_folder_name()
     os.makedirs(temp_dir)
     return temp_dir
+
+def get_file_lock(path: string, timeout=-1, stale=10):
+    class SimpleFileLock(filelock.SoftFileLock):
```

Comment:
This creates a new class every time the function gets called, which is unnecessary.